src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaUtils.java

```diff
@@ -0,0 +1,110 @@
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.search.SearchHit;
+
+import io.fabric8.elasticsearch.plugin.OpenshiftRequestContextFactory.OpenshiftRequestContext;
+import io.fabric8.elasticsearch.plugin.PluginClient;
+import io.fabric8.elasticsearch.plugin.PluginSettings;
+import io.fabric8.elasticsearch.plugin.model.Project;
+
+public class KibanaUtils {
+    
+    private static final Logger LOGGER = Loggers.getLogger(KibanaUtils.class);
+    private static final String INDICIES_TYPE = "index-pattern";
+    public static final Project ALL_ALIAS = new Project(".all", null);
+    private static final Project OPERATIONS = new Project(".operations", null);
+    public static final Project EMPTY_PROJECT = new Project(".empty-project", null);
+            
+    private final PluginClient pluginClient;
+    private String projectPrefix;
+    private final Pattern reIndexPattern;
+
+    public KibanaUtils(final PluginSettings settings, final PluginClient pluginClient) {
+        this.pluginClient = pluginClient;
+        this.projectPrefix = StringUtils.isNotBlank(settings.getCdmProjectPrefix()) ? settings.getCdmProjectPrefix() : "";
+        this.reIndexPattern = Pattern.compile("^" + projectPrefix + "\\.(?<name>[a-zA-Z0-9-]*)\\.(?<uid>.*)\\.\\*$");
+    }
+    
+    /**
+     * Determine a set of projects based on the index patterns that 
+     * were generated by this plugin.  Any index patterns that are user
+     * generated which do not match the format 'project.$NAME.$UID.*'
+     * are ignored
+     * 
+     * @param context   The OpenshiftRequestContext for this request
+     * @return a set of projects
+     */
+    public Set<Project> getProjectsFromIndexPatterns(OpenshiftRequestContext context) {
+        Set<Project> patterns = new HashSet<>();
+        SearchResponse response = pluginClient.search(context.getKibanaIndex(), INDICIES_TYPE);
+        if (response.getHits() != null && response.getHits().getTotalHits() > 0) {
+            for (SearchHit hit : response.getHits().getHits()) {
+                String id = hit.getId();
+                Project project = getProjectFromIndexPattern(id);
+
+                if (!project.getName().equals(id) || project.equals(ALL_ALIAS)) {
+                    patterns.add(project);
+                }
+
+                // else we found a user created index-pattern. Ignore
+            }
+        } else {
+            LOGGER.debug("No index-mappings found in the kibana index '{}'", context.getKibanaIndex());
+        }
+
+        return patterns;
+    }
+    
+    public Project getProjectFromIndexPattern(String index) {
+
+        if (StringUtils.isEmpty(index)) {
+            return Project.EMPTY;
+        }
+
+        Matcher matcher = reIndexPattern.matcher(index);
+        if (matcher.matches()) {
+            return new Project(matcher.group("name"), matcher.group("uid"));
+        }
+        return new Project(index, null);
+    }
+    
```

Comment:
if the index is `project.*`, then index = `project.*` and uid = `null`
then `formatIndexPattern` will return `project.project.*.*`?

Reply:
ok - hard to tell from the flow - but if you are sure that this method is never called with "project.*" then that's good enough for me